src/main/kotlin/io/arrowkt/tagless/repository/interpreter/AccountRepository.kt

```diff
@@ -0,0 +1,43 @@
+package io.arrowkt.tagless.repository.interpreter
+
+import arrow.Kind
+import arrow.core.*
+import arrow.typeclasses.MonadError
+import io.arrowkt.tagless.model.Account
+import io.arrowkt.tagless.model.Balance
+import io.arrowkt.tagless.repository.AccountRepository
+import io.arrowkt.tagless.today
+import java.time.LocalDate
+
+@Suppress("DELEGATED_MEMBER_HIDES_SUPERTYPE_OVERRIDE")
+class AccountRepositoryInMemory<F>(me: MonadError<F, Throwable>) : AccountRepository<F>,
+    MonadError<F, Throwable> by me {
+
+    private val repo by lazy { mutableMapOf<String, Account>() }
+
+    override fun query(no: String): Kind<F, Option<Account>> =
+        just(repo[no].toOption())
+
+    override fun store(a: Account): Kind<F, Account> {
+        repo[a.no] = a
```

Comment:
Side-effect. Is there any reason why MonadError and not MonadDefer?

Reply:
It is what the Scala version uses. Maybe MonadError in cats is lazy?